extensions/wikia/Discussions/maintenance/SiteStatus.class.php

```diff
@@ -0,0 +1,411 @@
+<?php
+
+namespace Discussions;
+
+
+class SiteStatus {
+	const PROD_SERVICE_URL = 'https://serivces.wikia.com';
+	const DEV_SERVICE_URL = 'https://services.wikia-dev.com';
+
+	const TIME_OUT = 5;
+
+	const TABLE_STATUS = 'discussion_migration.site_status';
+	const TABLE_CITY_LIST = 'wikicities.city_list';
+	const TABLE_COMMENTS = 'comments_index';
+	const TABLE_PAGE = 'page';
+
+	const STATUS_COMPLETE = 'complete';
+	const STATUS_RETRY = 'retry';
+	const STATUS_FAILED = 'failed';
+	const STATUS_NEW = 'new';
+
+	// How log we consider records "fresh" and not in need of a test
+	const CATALOG_TTL = 24*60*60;
+
+	private $test;
+	private $verbose;
+
+	/** @var \DatabaseBase */
+	private $localDbh;
+	/** @var \DatabaseBase */
+	private $centralDbr;
+	/** @var \DatabaseBase */
+	private $centralDbw;
+
+	private $siteId;
+	private $dbName;
+
+	private $hasStatusRecord = false;
+
+	/** @var \stdClass */
+	private $cityInfo;
+
+	private $status;
+	private $siteAvailable;
+	private $discussionsEnabled;
+	private $navigationEnabled;
+	private $forumsEnabled;
+	private $numThreadedForumPosts;
+	private $numWikiForumPosts;
+
+	/** @var \DateTime */
+	private $discussionsFound;
+	/** @var \DateTime */
+	private $firstPostFound;
+	/** @var \DateTime */
+	private $lastMigrated;
+	/** @var \DateTime */
+	private $lastUpdated;
+
+	public static function run( \DatabaseBase $db, $test = false, $verbose = false, $params = [] ) {
+		$status = new SiteStatus(
+			$db,
+			$verbose,
+			$test,
+			$params[\RunOnCluster::PARAM_SITE_ID],
+			$params[\RunOnCluster::PARAM_DB_NAME]
+		);
+
+		// Don't do anything if this has been recently updated by this script
+		if ( $status->skipCataloging() ) {
+			return;
+		}
+
+		$status->update();
+		$status->writeBack();
+	}
+
+	public function __construct( $localDbh, $verbose, $test, $siteId, $dbName ) {
+		$this->localDbh = $localDbh;
+		$this->verbose = $verbose;
+		$this->test = $test;
+		$this->siteId = $siteId;
+		$this->dbName = $dbName;
+
+		$this->debug("== Checking status for site $siteId / $dbName");
+
+		$this->load();
+	}
+
+	public function debug( $msg ) {
+		if ( empty( $this->verbose ) ) {
+			return;
+		}
+
+		echo $msg."\n";
+	}
+
+	private function load() {
+		( new \WikiaSQL() )
+			->SELECT( "*" )
+			->FROM( self::TABLE_STATUS )
+			->WHERE( 'site_id' )->EQUAL_TO( $this->siteId )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "site_id" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)